### PR TITLE
added authored date and settings tab

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR adds an authored date to article.

#### To Do

- [x] Tests

### CODE REVIEW

File: Document/ArticleDocument.php [hunk 1]
```diff
@@ -399,4 +406,28 @@ public function getPublished()
...
     }
+
+    /**
+     * Set authored date-time.
+     *
+     * @param \DateTime $authored
+     *
+     * @return $this
+     */
+    public function setAuthored($authored)
```
Comment (alexander-schranz) [inline Document/ArticleDocument.php]: If this is not able to be add the following line to __construct:

```
$this->authored = new \DateTime();
```

**Comment by alexander-schranz** [comment]
it would also make sense to add an `authors` field with each an array of contact ids.

**Comment by alexander-schranz** [comment]
tests worked for me locally so it seems more a travis-ci setup problem.

File: Controller/ArticleController.php [hunk 2]
```diff
@@ -283,6 +288,15 @@ private function persistDocument($data, $document, $locale)
         );
     }
 
+    private function getAuthors(array $authors)
```
Comment (alexander-schranz) [inline Controller/ArticleController.php]: phpdocs missing

File: Document/ArticleDocument.php [hunk 3]
```diff
@@ -141,6 +141,20 @@ class ArticleDocument implements
...
+     * @var \DateTime
+     */
+    protected $authored;
+
+    /**
+     * Array of contact-ids which has authored this article.
+     *
+     * @var int[]
+     */
+    protected $authors;
```
Comment (alexander-schranz) [inline Document/ArticleDocument.php]: shouldnt we initiliazed this as `array` that it always return an `array`?

File: Controller/ArticleController.php [hunk 4]
```diff
@@ -283,6 +288,15 @@ private function persistDocument($data, $document, $locale)
...
+    {
+        if (0 === count($authors)) {
+            return [$this->getUser()->getContact()->getId()];
```
Comment (alexander-schranz) [inline Controller/ArticleController.php]: I would do this only on create. If he want to unset the author its not possible.

File: Document/ArticleDocument.php (hunk 3 above)
Comment (wachterjohannes) [inline Document/ArticleDocument.php]: fixed

File: Controller/ArticleController.php [hunk 5]
```diff
@@ -284,6 +289,22 @@ private function persistDocument($data, $document, $locale)
...
+     * Returns authors or current user.
+     *
+     * @param int[] $authors
+     *
+     * @return int[]
+     */
+    private function getAuthors(array $authors)
+    {
+        if (0 === count($authors)) {
+            return [$this->getUser()->getContact()->getId()];
```
Comment (alexander-schranz) [inline Controller/ArticleController.php]: I would only to this when the document is created not on edit when the author is removed.

**Comment by alexander-schranz** [comment]
missing translations: sulu_article.authored, sulu_article.authors, ...

**Comment by alexander-schranz** [comment]
event missing when only change authors. I think you need to throw tab dirty on `sulu.content_changed`

**Pull request merged**